src/pages/Protect/Protect.jsx

```diff
@@ -1,22 +1,78 @@
 import { Typography } from "@mui/material";
-import { VIEW } from "../../utils/constants";
+import { CONTRACT_ADDRESS, VIEW, WETH_ADDRESS } from "../../utils/constants";
 import "../pages.scss";
 import { Button } from "../../components";
 import { InputField, ViewLinks, TokenModal, ProtectDetails } from "../Sections";
 import { useNavigate } from "react-router-dom";
-import { useContext } from "react";
+import { useContext, useEffect } from "react";
 import { setBlockData } from "../../ContextAPI/ContextApi";
 import { useWeb3React } from "@web3-react/core";
+import {abi as ERC20_ABI} from '../../Blockchain/Abis/ERC20.json'
+import {abi as stopLoss_Contract_abi} from '../../Blockchain/Abis/Stoploss.json'
+
 const Protect = (props) => {
   const navigate = useNavigate();
 
   // getting context API
-  const { connect_to_user_wallet, setType } = useContext(setBlockData);
+  const { connect_to_user_wallet, setType,protectedAmount,setProtectedAmount, setMetamaskEvent,setstopLoos_Contract} = useContext(setBlockData);
 
+  /** active - user wallet status  , active will be true if the  site is connected with the user wallet.
+   *  account -  user wallet address.
+   *  libray - Web3 or ether .
+   */
+  const { active, account, library } = useWeb3React();
+  /**
+   * @function approve_weth - This function will approve a amount of WETH  that user want to Protect.
+   * @param {the value  of WETH that user want to approve} _value
+   * @param {the token  address of WETH} _token
+   */
+  const approve_weth = async (_value, _token) => {
```

Comment:
approve_weth -> approveWeth or approveWETH ?

Note: follow the same for Protect ETH const variable name.
i.e if you're choosing approveWeth here, then go with protectWeth in Confirm.jsx